Sources/OpenAI/Public/Models/Chat/Chat.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  File.swift
+//  
+//
+//  Created by Federico Vitale on 14/11/23.
+//
+
+import Foundation
+
+public struct Chat: Codable, Equatable {
+    public let role: Role
+    /// The contents of the message. `content` is required for all messages except assistant messages with function calls.
+    public let content: StringOrCodable<[ChatContent]>?
+    
+    /// The name of the author of this message. `name` is required if role is `function`, and it should be the name of the function whose response is in the `content`. May contain a-z, A-Z, 0-9, and underscores, with a maximum length of 64 characters.
+    public let name: String?
+    public let functionCall: ChatFunctionCall?
+    
+    public enum Role: String, Codable, Equatable {
+        case system
+        case assistant
+        case user
+        case function
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case role
+        case content
+        case name
+        case functionCall = "function_call"
+    }
+    
+    public init(role: Role, content codable: Codable? = nil ,name: String? = nil, functionCall: ChatFunctionCall? = nil) {
```

Comment:
```suggestion
    public init(role: Role, contents codable: Codable? = nil ,name: String? = nil, functionCall: ChatFunctionCall? = nil) {
```
Let's make it as well content**S** to have matching constructors and not confuse future users ☺️